waspc/src/Wasp/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -87,15 +93,37 @@ instance ToJSON ExternalAuthProviderTemplateData where
         "authProviderEnabled" .= _authProviderEnabled externalProviderTD
       ]
 
-genRouter :: AppSpec -> Generator FileDraft
-genRouter spec = do
+genRouter :: AppSpec -> Generator [FileDraft]
+genRouter spec =
+  sequence
+    [ genRouterTsx spec,
```

Comment:
I suppose you meant `createRouterTemplateData` being called twice, and I agree, I'll try to organise stuff so we can avoid calling it twice 👍

Reply:
As for everything using the same definition, I haven't figured it out 🤷

Reply:
> We usually call this file (and its function) index.ts and genIndexTs and place it inside the router dir. How come we didn't do that here?

What about this part?

Reply:
> As for everything using the same definition, I haven't figured it out 🤷

This is fine since we're generating code. If we ever decide to improve it, a good question to ask is "If this was an app I'm writing, how would I handle route duplication" and then generate that. My guess is probably a `routes` dictionary with snake-case names converted to camelCase names. The same dict could also contain params.

From that, I think we could generate everything else (including the types). But no need to do it now IMO.

Reply:
> > We usually call this file (and its function) index.ts and genIndexTs and place it inside the router dir. How come we didn't do that here?

> What about this part?

I've just renamed the file from `router.jsx` to `router.tsx` so I can't speak to the previous file organisation 🤷 this feels okay 

> From that, I think we could generate everything else (including the types). But no need to do it now IMO.

I've done this in the latest iteration 👍 thanks for the push